src/Microsoft.TemplateEngine.Edge/Settings/TemplatePackageManager.cs

```diff
@@ -235,7 +237,20 @@ private async Task<TemplateCache> UpdateTemplateCacheAsync(bool needsRebuild, Ca
             Task<IReadOnlyList<ITemplatePackage>> getTemplatePackagesTask = GetTemplatePackagesAsync(needsRebuild, cancellationToken);
             if (!(_userTemplateCache is TemplateCache cache))
             {
-                cache = new TemplateCache(JObject.Parse(_paths.ReadAllText(_paths.TemplateCacheFile, "{}")), _logger);
+                try
+                {
+                    using (var fileReadStream = _environmentSettings.Host.FileSystem.OpenRead(_paths.TemplateCacheFile))
+                    using (var textReader = new StreamReader(fileReadStream))
+                    using (var jsonReader = new JsonTextReader(textReader))
+                    {
+                        _userTemplateCache = cache = new TemplateCache(JObject.Load(jsonReader), _logger);
+                    }
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogDebug(ex, "Failed to load templatecache.json.");
```

Comment:
should it be warning?

Reply:
Reason this is not warning, but just Debug is... it can happen that while reading cache exception can happen because logic, instead of doing looping and all that stuff, in worst case we will just re-create cache, so its not a big issue, just some perf hit... Hence Debug and not Warning.